packages/kbn-expandable-flyout/src/hooks/use_left_panel.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { rootKey } from './root_key';
+import { useUrlState } from './use_url_state';
+
+type LeftPanelState = any;
+
+export const useLeftPanel = () => {
+  const [leftPanelState, setLeftPanelState] = useUrlState(rootKey, 'leftPanel');
+
+  return [leftPanelState as LeftPanelState, setLeftPanelState] as const;
```

Comment:
what do you think about returning the data differently? Having an array like this forces anyone that wants to use `setLeftPanelState` to have an unused first variable (like you have in the [context](https://github.com/elastic/kibana/pull/169661/files#diff-0609a5c809720e697415e7696aeba92fe60d4e7f201def027e64b5afd0294dbbR94)) which isn't super friendly

Reply:
actually this is a good idea, at the time I was not sure what would be the best api

Reply:
these now return plain objects :)